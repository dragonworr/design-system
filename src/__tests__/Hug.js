import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Hug, Text, Card, theme } from '..'

const text = (
  <Text.span>
    Have you seen this <Text.span bold>card</Text.span> I‘m hugging?!
  </Text.span>
)

describe('Hug', () => {
  test('renders with border-radius from theme on top only', () => {
    const json = renderer.create(<Hug theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('border-top-left-radius', theme.radius)
    expect(json).toHaveStyleRule('border-top-right-radius', theme.radius)
    expect(json).not.toHaveStyleRule('border-bottom-left-radius', theme.radius)
    expect(json).not.toHaveStyleRule('border-bottom-right-radius', theme.radius)
  })

  test('renders text, icon, and Child with its own border', () => {
    const json = renderer.create(
      <Hug showChildBorder icon="thumbsUp" text={text} theme={theme}>
        <Card p={3} bg="white" color="text" theme={theme}>
          I‘m a card within a hug!
        </Card>
      </Hug>
    ).toJSON
    expect(json).toMatchSnapshot()
  })

  test('renders text when string is provided', () => {
    const json = renderer.create(
      <Hug
        icon="thumbsUp"
        text="I've been known to hug a card from time to time"
        theme={theme}
      >
        <Card p={3} bg="white" color="text" theme={theme}>
          I‘m a card within a hug!
        </Card>
      </Hug>
    ).toJSON
    expect(json).toMatchSnapshot()
  })

  test('renders text when array of nodes is provided', () => {
    const json = renderer.create(
      <Hug icon="thumbsUp" text={[text, text]} theme={theme}>
        <Card p={3} bg="white" color="text" theme={theme}>
          I‘m a card within a hug!
        </Card>
      </Hug>
    ).toJSON
    expect(json).toMatchSnapshot()
  })
})
