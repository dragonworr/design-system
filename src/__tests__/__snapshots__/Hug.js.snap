// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hug renders text when array of nodes is provided 1`] = `[Function]`;

exports[`Hug renders text when string is provided 1`] = `[Function]`;

exports[`Hug renders text, icon, and Child 1`] = `[Function]`;

exports[`Hug renders with border-radius from theme 1`] = `
.c2 {
  background-color: #0a0;
  color: #fff;
}

.c4 {
  padding: 32px;
  text-align: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 14px;
}

.c1 {
  border: 1px solid #0a0;
}

.c0 {
  border-radius: 2px;
}

<div
  className="c0 c1 c2"
  color="white"
>
  <div
    className="c3 c4"
  >
    <span
      className="c5"
      fontSize={1}
    />
  </div>
</div>
`;
